### PR TITLE
Fix for Enum.Format change in behavior between .Net Framework and .Net 5

### LINKED ISSUE #56200: Enum.Format change in behavior between .Net Framework and .Net 5
Repro:
```csharp
using System;

class Test
{
    [Flags]
    public enum TestEnum { None = 0, Item1 = 1, Item2 = 2, }

    static void Main()
    {
        Console.WriteLine(Enum.Format(typeof(TestEnum), (int)3, "G"));
    }
}
```
.NET Framework: `Item1, Item2`
.NET Core: `3`

---
### Description

While upgrading a project from .Net Framework 4.7.2 to .Net 5 and I noticed a difference in behavior of the EnumConverter.ConvertToString() method when passing an integer value that could represent a combination of flags enum values. Here is some code that reproduces the problem:

	class Program
	{
		[Flags]
		public enum TestEnum { None = 0, Item1 = 1, Item2 = 2, }
                
		static void Main(string[] args)
		{
			var none = TestEnum.None;
			var single = TestEnum.Item1;
			var multiple = TestEnum.Item1 | TestEnum.Item2;

			Console.WriteLine(none);
			Console.WriteLine(single);
			Console.WriteLine(multiple);

			var converter = new EnumConverter(typeof(TestEnum));
			Console.WriteLine(converter.ConvertToString(none));
			Console.WriteLine(converter.ConvertToString(single));
			Console.WriteLine(converter.ConvertToString(multiple));

			Console.WriteLine(converter.ConvertToString((int)none));
			Console.WriteLine(converter.ConvertToString((int)single));
			Console.WriteLine(converter.ConvertToString((int)multiple));
			Console.ReadLine();
		}
	}

Creating a simple console app with the code above, if I compile and run the app against .Net Framework 4.8 the last line outputs: 
`Item1, Item 2`
If I compile and run against .Net 5 the last line outputs:
`3`

I tried to have a look through the code here, compared to reference source and I see quite a few differences in there, but couldn't quite nail down the path that the code is taking to just return the int value as a string rather than calculating the enum value and converting that.

### PR DESCRIPTION
**Pull request by ChrisVicary**
Use InternalFormat when formatting an integer
value as an Enum that uses the FlagsAttribute.

Fix #56200

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=56261) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=56261) before we can accept your contribution.<br/><br/>:x: ChrisVicary [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=56261)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=56261) it.</sub>

**Review by jeffhandley — APPROVED** [review]
Thanks for submitting this fix!

**Review by stephentoub — APPROVED** [review]
LGTM. Thanks for the fix.

**Comment by jeffhandley** [comment]
@ChrisVicary It looks like you need to sign our Contributor License Agreement; that will need to be completed before we can merge this PR. Please let us know if you have any questions or concerns about signing it. [This comment](https://github.com/dotnet/runtime/pull/56261#issuecomment-885999435) has the links you need to complete that step.

**Comment by ChrisVicary** [comment]
@jeffhandley thanks for letting me know. It shouldn't be a problem, but I'm just waiting for legal approval from my employers. In hindsight I should have done that before submitting the PR, but hopefully it will not take long.

**Comment by jeffhandley** [comment]
Thanks for the update, @ChrisVicary; no worries at all--totally fine to complete it after your first PR is created.

**Comment by danmoseley** [comment]
Marking no merge meantime. But we do want it :)

**Comment by ChrisVicary** [comment]
@jeffhandley I have been approved, and have signed the CLA.

**Comment by jeffhandley** [comment]
Thank you, @ChrisVicary!!

**Comment by danmoseley** [comment]
Thanks for your first contribution, @ChrisVicary - we would welcome other contributions if you are interested. There are many issues open to choose from.

**Pull request merged**
